onedal/datatypes/data_conversion.cpp

```diff
@@ -233,36 +235,42 @@ static PyObject *convert_to_numpy_impl(const dal::array<T> &array,
 }
 
 template <int NpType, typename T>
-static PyObject *convert_to_py_from_csr_impl(const detail::csr_table &table) {
+static PyObject *convert_to_py_from_csr_impl(const csr_table &table) {
     PyObject *result = PyTuple_New(3);
     const std::int64_t rows_indices_count = table.get_row_count() + 1;
 
-    const std::int64_t *row_indices_one_based = table.get_row_indices();
-    std::uint64_t *row_indices_zero_based_data =
+    const std::int64_t *row_offsets_one_based = table.get_row_offsets();
+    std::uint64_t *row_offsets_zero_based_data =
         detail::host_allocator<std::uint64_t>().allocate(rows_indices_count);
     for (std::int64_t i = 0; i < rows_indices_count; ++i)
-        row_indices_zero_based_data[i] = row_indices_one_based[i] - 1;
+        row_offsets_zero_based_data[i] = row_offsets_one_based[i] - 1;
 
     auto row_indices_zero_based_array =
-        dal::array<std::uint64_t>::wrap(row_indices_zero_based_data, rows_indices_count);
+        dal::array<std::uint64_t>::wrap(row_offsets_zero_based_data, rows_indices_count);
     PyObject *py_row =
         convert_to_numpy_impl<NPY_UINT64, std::uint64_t>(row_indices_zero_based_array,
                                                          rows_indices_count);
     PyTuple_SetItem(result, 2, py_row);
 
-    const std::int64_t non_zero_count = row_indices_zero_based_data[rows_indices_count - 1];
-    const T *data = reinterpret_cast<const T *>(table.get_data());
+    const std::int64_t non_zero_count = table.get_non_zero_count();
+    const T *data = table.get_data<T>();
     auto data_array = dal::array<T>::wrap(data, non_zero_count);
 
     PyObject *py_data = convert_to_numpy_impl<NpType, T>(data_array, non_zero_count);
     PyTuple_SetItem(result, 0, py_data);
 
-    const std::int64_t *column_indices_one_based = table.get_column_indices();
-    std::uint64_t *column_indices_zero_based_data =
-        detail::host_allocator<std::uint64_t>().allocate(non_zero_count);
-    for (std::int64_t i = 0; i < non_zero_count; ++i)
-        column_indices_zero_based_data[i] = column_indices_one_based[i] - 1;
-
+    std::uint64_t *column_indices_zero_based_data = nullptr;
```

Comment:
This data type was not introduced in this PR, it was there before. It describes the data type of the indices passed to SciPy.
SciPy actually stores indices in signed data types: either npy_int32 or npy_int64 ([link](https://github.com/scipy/scipy/blob/main/scipy/sparse/sparsetools/util.h#L92), [link](https://github.com/scipy/scipy/blob/main/scipy/sparse/sparsetools/csr.cxx#L41)).

But it seems the code is working as it is now and I do not want to touch it in case we do not see the issues, in order not to break something.